sdk/tables/azure-data-tables/CHANGELOG.md

```diff
@@ -6,6 +6,7 @@
 * Added to support custom encoder in entity CRUD operations.
 * Added to support custom Entity type.
 * Added to support Entity property in Tuple and Enum types.
+* Added support for Microsoft Entra auth with Azure Cosmos DB for Table's OAuth scope (`https://cosmos.azure.com/.default`).
```

Comment:
We will also need to add a test for this - perhaps @YalinLi0312 can help get that added and recorded for the PR.

Reply:
@annatisch and @YalinLi0312, I'm trying to figure out how to write a unit tests to test that the correct scope is returned.

This is what I have so far but it's not working in Tox. It's likely some error on my end:

```python
from unittest.mock import Mock

from azure.data.tables._constants import STORAGE_OAUTH_SCOPE, COSMOS_OAUTH_SCOPE

from azure.data.tables._authentication import BearerTokenChallengePolicy

from _shared.asynctestcase import AsyncTableTestCase
from devtools_testutils import AzureRecordedTestCase, recorded_by_proxy
from preparers import tables_decorator
from _shared.testcase import TableTestCase

class TestConfigureCredential(AzureRecordedTestCase, TableTestCase):
    @tables_decorator
    @recorded_by_proxy
    def test_configure_credential_cosmos(self, tables_storage_account_name):
      token_credential = credential = Mock(spec_set=["get_token"], get_token=Mock(wraps=get_token))
      cosmos_endpoint = True
      expected = COSMOS_OAUTH_SCOPE
      policy = BearerTokenChallengePolicy(credential, "scope")
      result = policy._configure_credential(token_credential, cosmos_endpoint)
      assert result == expected
```

Reply:
We already have the tests, you can update these 2 in `test_table_client_cosmos.py` and `test_table_client_cosmos_async.py`.
https://github.com/Azure/azure-sdk-for-python/blob/f1be6511a73b1e419e326f6ebcc5c633986ecd1a/sdk/tables/azure-data-tables/tests/test_table_client_cosmos.py#L364-L430

Reply:
@annatisch, @kyle-patterson, & @YalinLi0312, I'm working on the unit tests in the latest commits, I'm just trying to find a way to get them to pass. I'm getting an "Authorization header doesn't conform to the required format" error message when running the `sdist` test suite. I'll take another look at it next week. If either of you see something obvious I missed, I'm happy to make changes.

Reply:
@seesharprun That's because of lacking permissions, we need to assign a new role for cosmos Entra Auth to use but there's no built-in role to use. We're talking with Cosmos team for that, not sure when the new role will come. For now, we need to use a custom role as a workaround. In this case, I'll handle the rest work. Thanks for your contribution again!